internal/xdscache/v3/snapshot.go

```diff
@@ -89,21 +94,40 @@ func (s *SnapshotHandler) GetCache() envoy_cache_v3.Cache {
 // Refresh is called when the EndpointSliceTranslator updates values
 // in its cache. It updates the EDS cache.
 func (s *SnapshotHandler) Refresh() {
-	version := uuid.NewString()
+	previouslyNotifiedResources := s.edsCache.GetResources()
+	currentResources := envoy_cache_v3.IndexRawResourcesByName(asResources(s.resources[envoy_resource_v3.EndpointType].Contents()))
+
+	toUpdate := make(map[string]envoy_types.Resource, len(currentResources))
+	toDelete := make([]string, 0, len(previouslyNotifiedResources))
+
+	for resourceName, previousResource := range previouslyNotifiedResources {
+		if newResource, ok := currentResources[resourceName]; ok {
+			// Add resources that were updated.
+			if !proto.Equal(newResource, previousResource) {
```

Comment:
we're effectively trading the req/resp expense for this comparison expense to ensure we don't send unnecessary updates